app/src/androidTest/java/org/simple/clinic/MigrationTestHelperWithForeignConstraints.kt

```diff
@@ -0,0 +1,36 @@
+package org.simple.clinic
+
+import android.arch.persistence.db.SupportSQLiteDatabase
+import android.arch.persistence.room.migration.Migration
+import android.arch.persistence.room.testing.MigrationTestHelper
+import android.support.test.InstrumentationRegistry
+import org.simple.clinic.di.AppSqliteOpenHelperFactory
+
+private const val TEST_DB_NAME = "migration-test"
+
+class MigrationTestHelperWithForeignConstraints : MigrationTestHelper(
+    InstrumentationRegistry.getInstrumentation(),
+    AppDatabase::class.java.canonicalName,
+    AppSqliteOpenHelperFactory()
+) {
+
+  fun createDatabase(version: Int): SupportSQLiteDatabase {
+    val db = super.createDatabase(TEST_DB_NAME, version)
+    db.setForeignKeyConstraintsEnabled(true)
+    return db
+  }
+
+  fun migrateTo(version: Int, vararg migrations: Migration): SupportSQLiteDatabase {
+    return super.runMigrationsAndValidate(TEST_DB_NAME, version, true, *migrations)
+  }
+
+  @Deprecated(message = "Use migrateTo() instead", replaceWith = ReplaceWith("helper.migrateTo(version, *migrations)"))
+  override fun runMigrationsAndValidate(
+      name: String?,
+      version: Int,
+      validateDroppedTables: Boolean,
+      vararg migrations: Migration?
+  ): SupportSQLiteDatabase {
+    return super.runMigrationsAndValidate(name, version, validateDroppedTables, *migrations)
```

Comment:
Instead of deprecating this, we should just throw an exception with a message to use `migrateTo` instead.

Reply:
Using this function isn't an error, why would you throw an exception?

Reply:
I don't think we should throw an exception here. Using this function is not an error. The only reason I deprecated this was because when you start typing `helper.migrate`, the person shouldn't see two migrate functions and get confused between them.